guis/panel/leak/panel_leak_utilities.py

```diff
@@ -92,3 +93,94 @@ def get_time(time_str):
         # df['PSI/degC'] = df["PRESSURE(PSI)"]/df["ROOM TEMPERATURE(C)"]
 
         return df
+
+
+################################################################################
+# READ INPUT
+# Read raw data from the database into a dataframe, structurally equivalent to that of ReadLeakRateFile, but with timestamps retained
+# TIME(DAYS)    FILLPSIA    RefPSIA PRESSURE(PSI)   BOX TEMPERATURE(C)  ROOM TEMPERATURE(C) Heater%
+################################################################################
+
+def readLeakDb(panel, tag):
+    leak_df = pd.DataFrame()
+    
+    #ensure that panel is type str
+    panel = str(panel)
+    
+    
+    con = sqlite3.connect('data/database.db')
+    cursor = con.cursor()
+
+    # acquire straw location
+    cursor.execute("SELECT * FROM straw_location WHERE number='"+str(panel)+"' AND location_type='MN'")
+    straw_location = str(cursor.fetchall()[0][0])
+    
+    # use straw location to acquire procedure
+    cursor.execute("SELECT * FROM procedure WHERE straw_location='"+straw_location+"' AND station='pan8'")
+    procedure = str(cursor.fetchall()[0][0])
+        
+    # use procedure and tag to acquire trial
+    cursor.execute("SELECT * FROM panel_leak_test_details WHERE procedure='"+procedure+"' AND tag='"+tag+"'")
+    trial = str(cursor.fetchall()[0][0])
+        
+    # use trial to acquire pertinent entries from measurement_panel_leak
+    leak_df = pd.read_sql_query("SELECT * FROM measurement_panel_leak WHERE trial='"+trial+"'", con)
+        
+    # rename pertinent columns
+    leak_df.rename(columns={"elapsed_days":"TIME(DAYS)"}, inplace=True)
+    leak_df.rename(columns={"pressure_diff":"PRESSURE(PSI)"}, inplace=True)
+    leak_df.rename(columns={"pressure_ref":"RefPSIA"}, inplace=True)
+    leak_df.rename(columns={"pressure_fill":"FillPSIA"}, inplace=True)
+    leak_df.rename(columns={"temp_box":"BOX TEMPERATURE(C)"}, inplace=True)
+    leak_df.rename(columns={"temp_room":"ROOM TEMPERATURE(C)"}, inplace=True)
+    leak_df.rename(columns={"heater_pct":"Heater%"}, inplace=True)
+    
+    # drop other columns
+    leak_df.drop(columns=['id', 'trial'], axis=1, inplace=True)
+        
+    con.close()
+        
+    return leak_df
+
+
+################################################################################
+# returns a list of leak test tags for an inputted panel
+################################################################################
+
+def getLeakTags(panel):
+    tag_list = []
+    con = sqlite3.connect('data/database.db')
+    cursor = con.cursor()
+    
+    # acquire straw location
```

Comment:
There are several functions lying around the repository that get a procedure ID and/or straw location. Should use those.

Reply:
I think you might have even worked with one or two of them here #147 or maybe #139?

If one doesn't exist in a common place, perhaps you could add it to `gui_utils.py`. I think that's a file.